### PR TITLE
New Text and Block options

### PR DESCRIPTION
**Pull request by swese44**
- Adds `ellipsis` option to Block
- Adds `maxWidth` (with ellipsis), `bold` and `uppercase` options to Text
- Adds a new demo story to show `maxWidth` in our message reply use case
- [x] Update MessageReply demo links after https://github.com/Microsoft/YamUI/pull/106 is merged

@unindented 2 things:
1. any way to remove placeholder .tsx files for the .md demo examples?
2. i think webpack/uglify is removing whitespace characters from JSX which messes up spacing between tags like Text/span, would be nice to disable that

* [x] Component `README.md` file is up-to-date.
* [x] Component is unit tested.

### CODE REVIEW

File: src/css/utility-classes.css [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  */
+
+.y-ellipsis {
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+  
+  /* overflow: hidden changes the baseline of inline-block elements, this fixes it */
+  vertical-align: bottom;
+}
```
Comment (unindented) [inline src/css/utility-classes.css]: Are `y-text*` classes also considered utility?

File: src/demos/MessageReply/MessageReply.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        <HorizontalListItem>
+          <Clickable><Icon icon="share" size={IconSize.XSMALL} /> Share</Clickable>
+        </HorizontalListItem>
+        <HorizontalListItem>
+          <Clickable>...</Clickable>
+        </HorizontalListItem>
+      </HorizontalList>
+    </Block>
+  </FixedGridColumn>
+</FixedGridRow>
```
Comment (unindented) [inline src/demos/MessageReply/MessageReply.md]: Why not move all this into the `.tsx` file?

File: src/css/utility-classes.css (hunk 1 above)
Comment (unindented) [reply src/css/utility-classes.css]: Ah sorry, that's happening in the other PR.

File: src/demos/MessageReply/MessageReply.md (hunk 2 above)
Comment (swese44) [reply src/demos/MessageReply/MessageReply.md]: good question. this way the full example is available for people to play with in the code panel.  would be nice if we didn't have to include a dummy tsx file

File: src/css/utility-classes.css (hunk 1 above)
Comment (swese44) [reply src/css/utility-classes.css]: merged so now showing up here :)

File: src/demos/MessageReply/MessageReply.md (hunk 2 above)
Comment (unindented) [reply src/demos/MessageReply/MessageReply.md]: I understand. The code for `Hovercard` demos currently lives in the `.tsx` file though...

Comment (swese44) [reply src/demos/MessageReply/MessageReply.md]: agree would be good to be consistent. let's think this through when we improve the examples/sections format

**Pull request merged**